pgjdbc/src/main/java/org/postgresql/Driver.java

```diff
@@ -545,90 +553,161 @@ public boolean jdbcCompliant() {
   public static Properties parseURL(String url, Properties defaults) {
     Properties urlProps = new Properties(defaults);
 
-    String urlServer = url;
-    String urlArgs = "";
-
-    int qPos = url.indexOf('?');
-    if (qPos != -1) {
-      urlServer = url.substring(0, qPos);
-      urlArgs = url.substring(qPos + 1);
-    }
-
-    if (!urlServer.startsWith("jdbc:postgresql:")) {
+    if (!url.startsWith("jdbc:postgresql:")) {
       LOGGER.log(Level.FINE, "JDBC URL must start with \"jdbc:postgresql:\" but was: {0}", url);
       return null;
     }
-    urlServer = urlServer.substring("jdbc:postgresql:".length());
-
-    if (urlServer.startsWith("//")) {
-      urlServer = urlServer.substring(2);
-      int slash = urlServer.indexOf('/');
-      if (slash == -1) {
-        LOGGER.log(Level.WARNING, "JDBC URL must contain a / at the end of the host or port: {0}", url);
-        return null;
-      }
-      urlProps.setProperty("PGDBNAME", URLCoder.decode(urlServer.substring(slash + 1)));
-
-      String[] addresses = urlServer.substring(0, slash).split(",");
-      StringBuilder hosts = new StringBuilder();
-      StringBuilder ports = new StringBuilder();
-      for (String address : addresses) {
-        int portIdx = address.lastIndexOf(':');
-        if (portIdx != -1 && address.lastIndexOf(']') < portIdx) {
-          String portStr = address.substring(portIdx + 1);
-          try {
-            int port = Integer.parseInt(portStr);
-            if (port < 1 || port > 65535) {
-              LOGGER.log(Level.WARNING, "JDBC URL port: {0} not valid (1:65535) ", portStr);
-              return null;
-            }
-          } catch (NumberFormatException ignore) {
-            LOGGER.log(Level.WARNING, "JDBC URL invalid port number: {0}", portStr);
+    url = url.substring("jdbc:postgresql:".length());
+    boolean ldapURL = false;
+    if (url.startsWith("@ldap")) {
+      ldapURL = true;
```

Comment:
ldapURL is not really necessary. 
Honestly pulling these out into two functions would make it much easier to read.

Reply:
Definitely true, I can create a parseJDBCURL and parseLDAPURL and have parseLDAPURL call the first one indeed. Agree with the naming ?

Reply:
Just use parseLDAP and parseJDBC. Thx